src/ucp/core/ucp_context.c

```diff
@@ -1074,16 +1081,27 @@ static ucs_status_t ucp_fill_resources(ucp_context_h context,
         context->tl_cmpts[i].cmpt = uct_components[i];
         context->tl_cmpts[i].attr.field_mask =
                         UCT_COMPONENT_ATTR_FIELD_NAME |
-                        UCT_COMPONENT_ATTR_FIELD_MD_RESOURCE_COUNT;
+                        UCT_COMPONENT_ATTR_FIELD_MD_RESOURCE_COUNT |
+                        UCT_COMPONENT_ATTR_FIELD_FLAGS;
         status = uct_component_query(context->tl_cmpts[i].cmpt,
                                      &context->tl_cmpts[i].attr);
         if (status != UCS_OK) {
             goto err_free_resources;
         }
 
+        if ((context->tl_cmpts[i].attr.flags & UCT_COMPONENT_FLAG_CM) &&
+            (context->config.ext.sockaddr_cm_enable == UCS_TRY)) {
+            context->config.ext.sockaddr_cm_enable = UCS_YES;
```

Comment:
Better not change the original field. have another boolean field to indicate if cm should be used or not in ucp_context_t::config